src/python/foglamp/device/server.py

```diff
@@ -69,20 +69,17 @@ class Server:
         cls._core_management_port = core_mgt_port
 
         try:
-            # TODO: Category name column is allows only 10 characters.
-            # This needs to be increased
             category = plugin
-
             config = {}
-
             await configuration_manager.create_category(category, config,
                                                         '{} Device'.format(plugin), True)
 
+            config = await configuration_manager.get_category_all_items(category)
+
             try:
-                plugin_module = config['plugin']
+                plugin_module = config['plugin']['value']
             except KeyError:
-                # TODO: Delete me
-                plugin_module = 'foglamp.device.coap_device'
+                sys.exit(1)
```

Comment:
Should we throw an exception instead?

Reply:
Let's at the very least log something. The worst thing is a service that mysteriously exists when this is at a customer site - the support guy will hate you for this :-)